lambdas/answer-validation/src/answer-validation-handler.ts

```diff
@@ -1,63 +1,155 @@
+      const input: AnswerValidationInputs =
+        this.safeRetrieveLambdaEventInputs(event);
+
+      logHelper.setSessionItemToLogging(input.sessionItem);
+      logHelper.setStatemachineValuesToLogging(input.statemachine);
+      logHelper.info(
+        `Handling request for session ${input.sessionItem.sessionId}`
+      );
+
+      if (
+        input.key === "rti-p60-earnings-above-pt" ||
+        input.key === "rti-p60-postgraduate-loan-deductions" ||
+        input.key === "rti-p60-student-loan-deductions" ||
+        input.key === "sa-income-from-pensions"
+      ) {
+        result = poundValidation(input);
+      } else if (input.key === "sa-payment-details") {
+        result = sAPaymentValidation(input);
+      } else {
+        result = penceValidation(input);
+      }
+      logHelper.info("Validation complete");
+
+      this.metricsProbe.captureMetric(
+        HandlerMetric.CompletionStatus,
+        MetricUnit.Count,
+        CompletionStatus.OK
+      );
+
+      return { validated: result };
+    } catch (error: any) {
+      const lambdaName = AnswerValidationHandler.name;
+      const errorText: string = error.message;
+
+      const errorMessage = `${lambdaName} : ${errorText}`;
+      logHelper.error(errorMessage);
+
+      this.metricsProbe.captureMetric(
+        HandlerMetric.CompletionStatus,
+        MetricUnit.Count,
+        CompletionStatus.ERROR
+      );
+
+      // Indicate to the statemachine a lambda error has occured
+      return { error: errorMessage };
+    }
+  }
+
+  private safeRetrieveLambdaEventInputs(event: any): AnswerValidationInputs {
+    if (!event) {
+      throw new Error("input event is empty");
+    }
+
+    const key = event.key;
+    if (!key) {
+      throw new Error("key not found");
+    }
+
+    const value = event.value;
+    if (!value) {
+      throw new Error("value not found");
+    }
+
+    // Session
+    const sessionItem = event?.sessionItem;
+    if (!sessionItem) {
+      throw new Error("Session item was not provided");
     } else {
-      result = penceValidation(event);
+      // Will throw errors on failure
+      SharedInputsValidator.validateUnmarshalledSessionItem(event.sessionItem);
     }
-    logger.info("The answers have been successfully validated");
-    return { validated: result };
+
+    // State machine values for logging
+    const statemachine = event.statemachine;
+    if (!statemachine) {
+      throw new Error("Statemachine values not found");
+    }
+
+    return {
+      sessionItem: sessionItem as SessionItem,
+      statemachine: statemachine as Statemachine,
+      key: key,
+      value: value,
+    } as AnswerValidationInputs;
   }
 }
-export function poundValidation(event: any): boolean {
-  if (event.value.match(/^([0-9]{0,12})$/)) {
-    logger.info("Validation in Pounds");
+export function poundValidation(input: AnswerValidationInputs): boolean {
+  if (input.value.match(/^([0-9]{0,12})$/)) {
+    logHelper.info("Validation in Pounds");
```

Comment:
Should this state validation success as the alternative is noted as failure. Just for clarity